src/main/java/com/google/devtools/build/lib/vfs/inmemoryfs/InMemoryContentInfo.java

```diff
@@ -120,6 +120,22 @@ public long getNodeId() {
     return System.identityHashCode(this);
   }
 
+  @Override
+  public synchronized int getPermissions() {
```

Comment:
I don't see what this needs to be synchronized with. Also, please make it `final`.

Reply:
Made it `final` and removed `synchronized`. I am not entirely sure about the intended behavior of `getPermissions` when the file permissions are updated concurrently, but I agree that `synchronized` doesn't help as the setters aren't synchronized.